examples/k8s-health-check/proxy_with_http_health_check.yaml

```diff
@@ -120,23 +139,31 @@ spec:
           # restarted.
           failureThreshold: 5
         readinessProbe:
+          # The /readiness probe returns OK when the proxy can establish
+          # a new connections to its databases.
+          #
+          # We do not recommend adding a readiness probe to the proxy sidecar
```

Comment:
If we don't recommend adding a readiness probe, should we just remove its current implementation and instead just check that the internal state is valid or something similar?

Reply:
I think it's OK to leave it in. A minority of customers will still want to use it: probably customers installing the proxy alongside existing legacy applications that don't have health checks.

Reply:
Also, the readiness check is helpful for our own e2e tests - because it ensures that the proxy can connect to the database server.

Reply:
Let's lighten the recommendation to something like, "Note: a readiness probe should be used carefully to ensure no unnecessary interruption to the application." And then give examples of how max connections will affect the readiness probe. WDYT?

Reply:
Ok, good idea. I'll put most of that nuance in README.md instead of a comment in this yaml file.

Reply:
That seems fine. I just don't want to be recommending against a feature we built. If we're recommending against it, we should be fixing that.